tests/integration/devfile/utils/utils.go

```diff
@@ -327,3 +327,65 @@ func validateContainerExecListDir(odoV1Watch OdoV1Watch, odoV2Watch OdoV2Watch,
 
 	return nil
 }
+
+// ExecCommand executes odo exec with a command
+func ExecCommand(context, cmpName string) {
+	args := []string{"create", "nodejs", cmpName, "--context", context}
+	helper.CmdShouldPass("odo", args...)
+
+	helper.CopyExample(filepath.Join("source", "devfiles", "nodejs", "project"), context)
+	helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", "devfile.yaml"), filepath.Join(context, "devfile.yaml"))
+
+	args = []string{"push", "--context", context}
+	helper.CmdShouldPass("odo", args...)
+
+	args = []string{"exec", "--context", context}
+	args = append(args, []string{"--", "touch", "blah.js"}...)
+	helper.CmdShouldPass("odo", args...)
+}
+
+//ExecWithoutCommand executes odo exec with no user command and fails
+func ExecWithoutCommand(context, cmpName string) {
+	args := []string{"create", "nodejs", cmpName, "--context", context}
+	helper.CmdShouldPass("odo", args...)
+
+	helper.CopyExample(filepath.Join("source", "devfiles", "nodejs", "project"), context)
+	helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", "devfile.yaml"), filepath.Join(context, "devfile.yaml"))
+
+	args = []string{"push", "--context", context}
+	helper.CmdShouldPass("odo", args...)
+
+	args = []string{"exec", "--context", context}
+	args = append(args, "--")
+	output := helper.CmdShouldFail("odo", args...)
+
+	Expect(output).To(ContainSubstring("no command was given"))
+
+}
+
+//ExecWithInvalidCommand executes odo exec with a invalid command
+func ExecWithInvalidCommand(context, cmpName, pushTarget string) {
```

Comment:
In my opinion this is very specific function which created nodejs component -> pushes it and then exec a command, can we please consider breaking it down into
- `CreateAndPush(componentType, cmpName, context string)` that can be called with `nodejs` as componentType and so on...
- `ExecCommand(context string)` which execs command and returns the output

Reply:
But most of the tests are written in the same format.